polytorch/metrics.py

```diff
@@ -9,38 +9,148 @@ def get_predictions_target_for_index(predictions, *targets, data_index=None, fea
     return function(my_predictions, my_targets)
 
 
-def categorical_accuracy(predictions, *targets, data_index=None, feature_axis=-1):
+def categorical_accuracy(predictions, *targets, data_index=None, feature_axis=-1) -> torch.Tensor:
     my_predictions, my_targets = get_predictions_target_for_index(predictions, *targets, data_index=data_index, feature_axis=feature_axis)
     my_predictions = torch.argmax(my_predictions, dim=feature_axis)
 
     accuracy = (my_predictions == my_targets).float().mean()
     return accuracy
 
 
-def mse(predictions, *targets, data_index=None, feature_axis=-1):
+def binary_accuracy(predictions, *targets, data_index=None, feature_axis=-1) -> torch.Tensor:
+    my_predictions, my_targets = get_predictions_target_for_index(predictions, *targets, data_index=data_index, feature_axis=feature_axis)
+    my_predictions = my_predictions >= 0.0
+    my_predictions = squeeze_prediction(my_predictions, my_targets, feature_axis)
+
+    accuracy = (my_predictions == my_targets).float().mean()
+    return accuracy
+
+
+def mse(predictions, *targets, data_index=None, feature_axis=-1) -> torch.Tensor:
     return function_metric(predictions, *targets, data_index=data_index, feature_axis=feature_axis, function=F.mse_loss)
 
 
-def l1(predictions, *targets, data_index=None, feature_axis=-1):
+def l1(predictions, *targets, data_index=None, feature_axis=-1) -> torch.Tensor:
     return function_metric(predictions, *targets, data_index=data_index, feature_axis=feature_axis, function=F.l1_loss)
 
 
-def smooth_l1(predictions, *targets, data_index=None, feature_axis=-1):
+def smooth_l1(predictions, *targets, data_index=None, feature_axis=-1) -> torch.Tensor:
     return function_metric(predictions, *targets, data_index=data_index, feature_axis=feature_axis, function=F.smooth_l1_loss)
 
 
+def calc_dice_score(predictions, target, smooth:float=1.) -> torch.Tensor:
+    predictions = predictions.view(-1)
+    target = target.view(-1)
+    intersection = (predictions * target).sum()
+    
+    return ((2. * intersection + smooth) /
+              (predictions.sum() + target.sum() + smooth)
+    )
+
+
+def calc_generalized_dice_score(predictions, target, power:float=2.0, smooth:float=1.0, feature_axis:int=-1) -> torch.Tensor:
+    """
+    A generalized Dice score for multi-class segmentation.
+
+    If power=0.0, this is equivalent to normal Dice score (i.e. volume "implicit" weighting)
+    If power=1.0, this is equivalent to 'equal' weighting.
+    If power=2.0, this is equivalent to 'inverse volume' weighting.
+
+    See:
+        - https://www.sciencedirect.com/science/article/pii/S2590005619300049#bib73
+        - https://arxiv.org/pdf/1707.03237.pdf
+        - https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=1717643
+    """
+    n_classes = predictions.shape[feature_axis]
+    numerator = 0.0
+    denominator = 0.0
+    slice_indices = [slice(0, None)] * len(predictions.shape)
```

Comment:
Interesting use of `slice` here; seems like a way to use `:` when you don't know which dimensions you want to use it on.

Reply:
that's right, that's why I used it. I don't know a better way

Reply:
I think an alternative in numpy would be something like `np.take(a=predictions, indices=i, axis=feature_axis)`. But I think your approach is just as good if not better and I don't know if there is an equivalent of `np.take` in pytorch (maybe [`index_select`](https://pytorch.org/docs/stable/generated/torch.index_select.html#torch.index_select)?).